### ISSUE TITLE
Replace does not work in version 0.9

### ISSUE THREAD

**Issue by johnblommers** [vkbo/novelWriter#327]
On Ubuntu 19.10 we launch novelWriter on the command line, open a document and try  to replace the word `Van` with `VanDer`. So we:

1. Type <kbd>Ctrl</kbd><kbd>f</kbd> which brings up the search and replace pane
1. Type `Van` into the search field
1. Click the disclosure triangle
1. Type `VanDer` into the replace field
1. Click the search icon. novelWriter highlights the first occurrence of `Van`.
1. Click the replace icon. Nothing happens.

At the same time, in the console, we see a clue:

```
Traceback (most recent call last):
  File "/home/johnb/Applications/novelWriter/nw/gui/doceditor.py", line 1595, in _doReplace
    self.docEditor.docAction(nwDocAction.REPL_NEXT)
  File "/home/johnb/Applications/novelWriter/nw/gui/doceditor.py", line 535, in docAction
    self._replaceNext()
  File "/home/johnb/Applications/novelWriter/nw/gui/doceditor.py", line 1240, in _replaceNext
    if not theCursor.hasSelection():
UnboundLocalError: local variable 'theCursor' referenced before assignment
```

**Comment by vkbo** [comment]
Dammit. I checked that before merging everything, but probably broke it in the final code cleanup. I forgot to recheck it after. I really should add a test for the search/replace tool.

**Comment by vkbo** [comment]
Fixed. It was just a single line of code that needed to be moved up a few lines.

**Comment by johnblommers** [comment]
Sweet!

**Comment by vkbo** [comment]
This is a bad enough bug to warrant a patch release, but I'll hold off on it for a couple of hours to see if you find anything else.

I've been using this version of novelWriter for my own project both yesterday and today, and fixed a couple of annoyances before the release, but haven't found any other bugs.

**Comment by johnblommers** [comment]
Your patch worked nicely. Search and replace works exactly as you intend it to. I know you don't want it to blindly do a complete search and replace in one click and that's actually OK because it makes it hard to make a mess if the operation goes south.

No more issues revealed themselves  to me.

**Comment by vkbo** [comment]
Ok, I'll release the patch then.

I think the search/replace tool has a decent balance between features and staying within my more or less settled rule of avoiding bulk editing of documents.

I spent quite some time designing the tool to be compact and still have the information easily available. I have tended to avoid task bars in favour of menus (less clutter), but in this case I think the task bar works better.
